website/docs/d/supervisor_namespace.html.markdown

```diff
@@ -3,19 +3,56 @@ layout: "vcfa"
 page_title: "VMware Cloud Foundation Automation: vcfa_supervisor_namespace"
 sidebar_current: "docs-data-source-vcfa-supervisor-namespace"
 description: |-
-  Provides a VMware Cloud Foundation Automation Supervisor Namespace data source.
+  Provides a data source to read a Supervisor Namespace from VMware Cloud Foundation Automation.
 ---
 
 # vcfa\_supervisor\_namespace
 
-Provides a VMware Cloud Foundation Automation Supervisor Namespace data source.
+Provides a data source to read a Supervisor Namespace from VMware Cloud Foundation Automation.
 
-## Example Usage
+_Used by: **Tenant**_
+
+-> This data source may use the [Kubernetes provider](https://registry.terraform.io/providers/hashicorp/kubernetes),
+to see how to obtain the Kubeconfig, please check the [`vcfa_kubeconfig`](/providers/vmware/vcfa/latest/docs/data-sources/kubeconfig) data source.
+
+## Example Usage - Retrieving from a Project resource
+
+```hcl
+# A project resource managed by the Kubernetes provider
+resource "kubernetes_manifest" "project" {
+  manifest = {
+    "apiVersion" = "project.cci.vmware.com/v1alpha2"
+    "kind"       = "Project"
+    "metadata" = {
+      "name" = "tf-demo-project"
+    }
+    "spec" = {
+      "description" = "Created by Terraform VCFA Provider"
+    }
+  }
+}
+
+data "vcfa_supervisor_namespace" "supervisor_namespace" {
+  name         = "tf-supervisor-namespace"
+  project_name = kubernetes_manifest.project.manifest["metadata"]["name"]
+}
+```
+
```

Comment:
Seemingly this example is a bit counter-intuitive. 
It creates a project, but at the same time it expects that a namespace is already in that project which can't exist without a project

Reply:
Very true, removed.